File: schemas/expenses.py
import datetime

from pydantic import BaseModel

class ExpenseIn(BaseModel):
    category_id: int
    amount: float
    occurred_at: datetime.datetime
    title: str
    note: str


class ExpenseOut(BaseModel):
    id: int
    user_id: int
    category_id: int
    amount: float
    occurred_at: datetime.datetime
    title: str
    note: str

    class Config:
        from_attributes = True

File: frontend/src/types/expenses.ts
export type Expense = {
  id: number;
  user_id: number;
  category_id: number;
  amount: number;
  occurred_at: string;
  title: string;
  note: string;
};

export type ExpensePayload = {
  category_id: number;
  amount: number;
  occurred_at: string;
  title: string;
  note: string;
};

export type ExpenseListResponse = {
  items: Expense[];
  meta: {
    page: number;
    limit: number;
    total: number;
    pages: number;
  };
};
